src/txHashBuilder.c

```diff
@@ -1180,10 +1183,46 @@ static void txHashBuilder_assertCanLeaveMint(tx_hash_builder_t* builder)
 	ASSERT(builder->multiassetData.remainingTokens == 0);
 }
 
+void txHashBuilder_addNetworkId(tx_hash_builder_t* builder, uint8_t networkId)
+{
+	_TRACE("state = %d", builder->state);
+
+	txHashBuilder_assertCanLeaveMint(builder);
+
+	// add fee item into the main tx body map
+	BUILDER_APPEND_CBOR(CBOR_TYPE_UNSIGNED, TX_BODY_KEY_NETWORK_ID);
+	BUILDER_APPEND_CBOR(CBOR_TYPE_UNSIGNED, networkId);
+
+	builder->state = TX_HASH_BUILDER_IN_NETWORK_ID;
+}
+
+static void txHashBuilder_assertCanLeaveNetworkId(tx_hash_builder_t* builder)
+{
+	_TRACE("state = %d", builder->state);
+
+	switch (builder->state) {
+	case TX_HASH_BUILDER_IN_NETWORK_ID:
+		break;
+
+	case TX_HASH_BUILDER_IN_MINT:
+	case TX_HASH_BUILDER_IN_VALIDITY_INTERVAL_START:
+	case TX_HASH_BUILDER_IN_AUX_DATA:
+	case TX_HASH_BUILDER_IN_WITHDRAWALS:
+	case TX_HASH_BUILDER_IN_CERTIFICATES:
+	case TX_HASH_BUILDER_IN_TTL:
+	case TX_HASH_BUILDER_IN_FEE:
+		txHashBuilder_assertCanLeaveMint(builder);
+		ASSERT(!builder->includeValidityIntervalStart);
+		break;
```

Comment:
I'm confused by this. Is it true it will disappear if we make network id mandatory?

Reply:
Well, it would look like this: ASSERT(builder->state == TX_HASH_BUILDER_IN_NETWORK_ID); since there is no other way to leave it. Not much of a code reduction, but still some bytes.

Reply:
I am mostly interested in the simplicity :)
Having `ASSERT(!builder->includeValidityIntervalStart);` in a function named `txHashBuilder_assertCanLeaveNetworkId` is just confusing to me, and it seems simpler to just get rid of it (by making network id always included) instead of thinking about why it is here, and whether it should be.

Reply:
There are already examples for this in other mandatory elements, like Fee. I think some kind of assert is needed, so the Confirm state can check if anything is missing.